src/main/java/io/cdap/plugin/spark/dynamic/ScalaSparkCompute.java

```diff
@@ -110,6 +111,13 @@ public void prepareRun(SparkPluginContext context) throws Exception {
       // If there is no output schema configured, derive it from the DataFrame
       // Otherwise, assume the DataFrame has the correct schema already
       outputSchema = DataFrames.toSchema((DataType) invokeDataFrameMethod(result, "schema"));
+    } else {
+      Schema dataSchema = DataFrames.toSchema((DataType) invokeDataFrameMethod(result, "schema"));
+      if (!dataSchema.isCompatible(outputSchema)) {
+        FailureCollector collector = context.getFailureCollector();
+        collector.addFailure("Schema mismatch.", "Output schema is not matching input schema.");
```

Comment:
Could you please rephrase:
`"Output schema is not compatible with input schema.", null`